### PR TITLE
feat: support delimiters in hive csv upload

### PR DESCRIPTION
**Pull request by serenajiang**
### SUMMARY
Previously, hive csv upload only supported `,` as a delimiter. This is confusing for users because the UI allows you to specify a delimiter.

Altered to allow other delimiters.

### TEST PLAN
Tested csv + tsv uploads.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [X] Introduces new feature or API
- [ ] Removes existing feature or API

@etr2460 @bkyryliuk @john-bodley

### CODE REVIEW

File: superset/db_engine_specs/hive.py [hunk 1]
```diff
@@ -163,10 +163,11 @@ def convert_to_hive_type(col_type: str) -> str:
             bucket_path,
             os.path.join(upload_prefix, table.table, os.path.basename(filename)),
         )
-        # TODO(bkyryliuk): support other delimiters
+
         sql = f"""CREATE TABLE {str(table)} ( {schema_definition} )
-            ROW FORMAT DELIMITED FIELDS TERMINATED BY ',' STORED AS
-            TEXTFILE LOCATION '{location}'
+            ROW FORMAT SERDE 'org.apache.hadoop.hive.contrib.serde2.MultiDelimitSerDe'
+            WITH SERDEPROPERTIES ("field.delim"="{csv_to_df_kwargs['sep']}")
```
Comment (serenajiang) [inline superset/db_engine_specs/hive.py:169]: Yes, this code is really insecure, but aren't we all? Hahahahahahahaha

Comment (bkyryliuk) [reply superset/db_engine_specs/hive.py:169]: good point, I would suggest to use params for this one: https://docs.sqlalchemy.org/en/13/core/connections.html#sqlalchemy.engine.Engine.execute

File: superset/db_engine_specs/hive.py [hunk 2]
```diff
@@ -163,13 +163,18 @@ def convert_to_hive_type(col_type: str) -> str:
             bucket_path,
             os.path.join(upload_prefix, table.table, os.path.basename(filename)),
         )
-        # TODO(bkyryliuk): support other delimiters
-        sql = f"""CREATE TABLE {str(table)} ( {schema_definition} )
-            ROW FORMAT DELIMITED FIELDS TERMINATED BY ',' STORED AS
-            TEXTFILE LOCATION '{location}'
+        sql = text(
```
Comment (serenajiang) [inline superset/db_engine_specs/hive.py:166]: Note `text` is used for language agnostic `:param`s.

File: superset/db_engine_specs/hive.py [hunk 3]
```diff
@@ -163,13 +163,18 @@ def convert_to_hive_type(col_type: str) -> str:
...
+            f"""CREATE TABLE {str(table)} ( {schema_definition} )
+            ROW FORMAT DELIMITED FIELDS TERMINATED BY :delim
+            STORED AS TEXTFILE LOCATION :location
             tblproperties ('skip.header.line.count'='1')"""
+        )
         engine = cls.get_engine(database)
-        engine.execute(sql)
+        engine.execute(
+            sql,
+            delim=csv_to_df_kwargs['sep'].encode().decode('unicode_escape'),
```
Comment (serenajiang) [inline superset/db_engine_specs/hive.py:175]: Required to escape `\\t` as `\t`.

**Comment by serenajiang** [comment]
@bkyryliuk Used params as suggested. Switched back to the original row format (single character delimiter) because unicode made it hard to use complex delimiters.

**Comment by serenajiang** [comment]
ping @bkyryliuk

**Review by bkyryliuk — APPROVED** [review]
@serenajiang looks good, please fix the linter

**Pull request merged**